src/helpers.ts

```diff
@@ -0,0 +1,21 @@
+import type { HonoRequest } from "hono";
+import type { SafeParseReturnType, ZodType, ZodTypeDef, output } from "zod";
+
+// biome-ignore lint/suspicious/noExplicitAny: <explanation>
+export async function requestBody<T extends ZodType<any, ZodTypeDef, any>>(
```

Comment:
@dahlia this makes working with JSON or FormData and validating it MUCH easier. The signature of `ZodType` is `ZodType<Output = any, Def extends ZodTypeDef = ZodTypeDef, Input = Output>` as we only have an schema and the input to `safeParseAsync` is either a record-like object or `any` (in the case of `req.json()`), I think the uses of `any` here are fine.

I could add explanations maybe?

But this function should help clean up all those cases where we had the if/else on content type to handle both json and formdata.